org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/staging/actions/OpenObjectAction.java

```diff
@@ -55,79 +37,14 @@ protected boolean updateSelection(IStructuredSelection selection) {
-		Shell shell = ((AbapGitStagingView) this.view).getSite().getShell();
-
-		if (GitCredentialsService.checkIfCredentialsRequired(this.view.getExternalRepositoryInfo())) {
-			if (getGitCredentialsAndValidate(shell, repo.getUrl()).equals(Status.CANCEL_STATUS)) {
-				//reset if user cancels refresh
-				this.credentials = null;
-				return;
-			}
-		} else {
 			openObjectInEditor();
-		}
-
 	}
 
 	private void openObjectInEditor() {
 		IStructuredSelection selection = (IStructuredSelection) this.treeViewer.getSelection();
 		for (Object object : selection.toList()) {
-			AbapGitUIServiceFactory.createAbapGitStagingService().openEditor(object, this.view.getProject(), this.credentials);
-		}
-	}
-
-	private void repositoryChecks(IProgressMonitor monitor, IExternalRepositoryInfoRequest credentials) {
-		//perform repository checks
-		if (AbapGitUIServiceFactory.createRepositoryService(this.view.getProject()).getURIFromAtomLink(this.view.getRepository(),
-				IRepositoryService.RELATION_CHECK) != null) {
-			monitor.beginTask(Messages.AbapGitStaging_check_job_title, IProgressMonitor.UNKNOWN);
-			IRepositoryService repoService = AbapGitUIServiceFactory.createRepositoryService(this.view.getProject());
-			repoService.repositoryChecks(monitor, credentials, this.view.getRepository());
-		}
-	}
-
-	private IStatus getGitCredentials(Shell shell, String repositoryURL, String errorMessage) {
-		//get credentials
-		this.credentials = GitCredentialsService.getCredentialsFromUser(shell, repositoryURL, errorMessage);
-		if (this.credentials == null) {
-			return Status.CANCEL_STATUS;
-		}
-		return Status.OK_STATUS;
-	}
-
-	private void handleException(Exception e, String repoURL, Shell shell) {
-		//invalid credentials : this condition check is only valid from 2002
-		if (e instanceof ResourceException && GitCredentialsService.isAuthenticationIssue((ResourceException) e)) {
-			ErrorHandlingService.handleExceptionAndDisplayInErrorDialog(e, shell);
-			if (getGitCredentials(shell, repoURL, e.getLocalizedMessage()).equals(Status.OK_STATUS)) {
-				validateCredsAndOpenObject(shell, repoURL);
-			}
-		} else {
-			ErrorHandlingService.openErrorDialog("Error", e.getLocalizedMessage(), shell, true); //$NON-NLS-1$
-		}
-	}
-
-	private void validateCredsAndOpenObject(Shell shell, String repositoryURL) {
-				try {
-					//perform repository checks
-			repositoryChecks(new NullProgressMonitor(), OpenObjectAction.this.credentials);
-			openObjectInEditor();
-				} catch (CommunicationException | ResourceException | OperationCanceledException e) {
-					handleException(e, repositoryURL, shell);
-
-				} catch (OutDatedClientException e) {
-					AdtUtilUiPlugin.getDefault().getAdtStatusService().handle(e, null);
-				}
-	}
-
-	private IStatus getGitCredentialsAndValidate(Shell shell, String repositoryURL) {
-		if (getGitCredentials(shell, repositoryURL, null).equals(Status.CANCEL_STATUS)) {
-			return Status.CANCEL_STATUS;
```

Comment:
`OpenEditor` action does not require credentials as it opens a local file/object in an editor.
Thus all the credentials related code is removed.